src/org/etools/j1939tools/modules/NOxBinningModule.java

```diff
@@ -48,35 +49,73 @@ public NOxBinningModule(DateTimeModule dateTimeModule) {
             "                         ", "NOx Mass, g", "NOx Mass, g", "EOE, kWh", "Fuel, l", "Hours, min", "Dist, km"
     };
 
-    private static final int[] ACTIVE_100_HOUR_PGNS = new int[] { 0xFB17, 0xFB16, 0xFB15, 0xFB12, 0xFB13, 0xFB14 };
+    private static final Integer[] ACTIVE_100_HOUR_PGNS = new Integer[] { 0xFB17, 0xFB16, 0xFB15, 0xFB12, 0xFB13,
+            0xFB14 };
 
-    private static final int[] STORED_100_HOUR_PGNS = new int[] { 0xFB11, 0xFB10, 0xFB0F, 0xFB0C, 0xFB0D, 0xFB0E };
+    private static final Integer[] STORED_100_HOUR_PGNS = new Integer[] { 0xFB11, 0xFB10, 0xFB0F, 0xFB0C, 0xFB0D,
+            0xFB0E };
 
-    private static final int[] LIFETIME_PGNS = new int[] { 0xFB0B, 0xFB0A, 0xFB09, 0xFB06, 0xFB07, 0xFB08 };
+    private static final Integer[] LIFETIME_PGNS = new Integer[] { 0xFB0B, 0xFB0A, 0xFB09, 0xFB06, 0xFB07, 0xFB08 };
 
-    private static final int[] ENGINE_LIFETIME_PGNS = new int[] { 0, 0, 0xFB05, 0xFB02, 0xFB03, 0xFB04 };
+    private static final Integer[] ENGINE_LIFETIME_PGNS = new Integer[] { 0, 0, 0xFB05, 0xFB02, 0xFB03, 0xFB04 };
```

Comment:
if (List.of(LIFETIME_PGNS).contains(Integer.valueOf(pkt.getPgnDefinition().getId()))) {
                packetsContainLifetime[0] = true;
            }
because that doesn't work if they are ints